src/main/java/games/strategy/net/nio/Decoder.java

```diff
@@ -62,6 +62,7 @@ private void loop() {
         try {
           data = reader.take();
         } catch (final InterruptedException e) {
+          Thread.currentThread().interrupt();
           continue;
```

Comment:
Not directly related to the update, but the `continue` looks very suspicious. AFAIK we should abort any ongoing operations on interrupted exception (user is trying to kill the program). Here I would expect to see a `running = false`, and a `break` from the current running loop.

Reply:
Agreed.  However, this might require a bit more research and testing to ensure there's no crazy assumption someplace that breaks by doing this.  Since this code is basically in the core of all network communication, I'd prefer to focus on that in a separate PR.  I've added it to my TODO list.

Reply:
@DanVanAtta Following up on this, it appears the expected behavior you described is already present.  `running` is set to `false` a few lines above in the `shutDown()` method immediately before interrupting the thread running `loop()`.  So the `continue` will jump to the top of the loop where it will exit because `running` is `false`.

However, I think the `InterruptedException` handler and one other exception handler can be simplified a bit to avoid such confusion in the future.  I'll submit a PR proposing those changes.